src/main/java/org/subethamail/smtp/server/Session.java

```diff
@@ -287,7 +306,8 @@ public void setSocket(Socket socket) throws IOException {
         this.socket = socket;
         this.input = this.socket.getInputStream();
         this.reader = new CRLFTerminatedReader(this.input);
-        this.writer = new PrintWriter(this.socket.getOutputStream());
+        this.output = this.socket.getOutputStream();
+        this.writer = new PrintWriter(this.output);
```

Comment:
This is a tad risky in that normally I would not suggest it is a good idea to write to both the writer and `output` because `PrintWriter` buffers. I checked the use of `writer` though and noticed it was followed immediately by `writer.flush`. This ensures the problem doesn't happen but I'd like to ensure that that flush command does not get removed in the future. Can you add a comment just before the `writer.flush` that mentions it has to stay there and why please.

Reply:
Adding comments. Btw without flush you can't know if data has been sent really to client so I don't think that it will be removed